executor/executor_test.go

```diff
@@ -254,17 +265,22 @@ func (s *ExecutorTestSuite) TestHandleLaunch() {
 	assert.Error(s.T(), s.executor.handleLaunch(&ev))
 	assert.Equal(s.T(), *mesos.TASK_FAILED.Enum(), *pullFirstUpdate(s.executor.UnackedUpdates).Status.State) // Should be a TASK_FAILED update
 
-	// Nominal case
-	s.executor.HookManager.PostRunHooks = []*hook.Hook{} // Remove previously added failing hook
-	assert.Nil(s.T(), s.executor.handleLaunch(&ev))      // Should return nil (launch successful)
-	assert.NotEmpty(s.T(), s.executor.UnackedTasks)      // Should not be empty (task waiting for acknowledgment)
-	assert.NotEmpty(s.T(), s.executor.ContainerTasks)    // Should not be empty (new running task for this container)
-
+	// Nominal case (long-running container)
+	s.executor.HookManager.PostRunHooks = []*hook.Hook{}                                                      // Remove previously added failing hooks
+	assert.Nil(s.T(), s.executor.handleLaunch(&ev))                                                           // Should return nil (launch successful)
+	assert.NotEmpty(s.T(), s.executor.UnackedTasks)                                                           // Should not be empty (task waiting for acknowledgment)
+	assert.NotEmpty(s.T(), s.executor.ContainerTasks)                                                         // Should not be empty (new running task for this container)
 	assert.NotEmpty(s.T(), s.executor.UnackedUpdates)                                                         // Should not be empty (TASK_RUNNING update)
-	assert.Equal(s.T(), *mesos.TASK_RUNNING.Enum(), *pullFirstUpdate(s.executor.UnackedUpdates).Status.State) // Should be a TASK_FAILED update
+	assert.Equal(s.T(), *mesos.TASK_RUNNING.Enum(), *pullFirstUpdate(s.executor.UnackedUpdates).Status.State) // Should be a TASK_RUNNING update
+	containerTask := s.executor.ContainerTasks[s.taskInfo.GetTaskID()]                                        // Get running task
+	assert.Equal(s.T(), "fakeContainerID", containerTask.ContainerID)                                         // Should be equal to the container ID
+	assert.Empty(s.T(), s.executor.UnackedUpdates)                                                            // Should be empty
+	close(done)                                                                                               // Simulate container exit
+	for len(s.executor.UnackedUpdates) == 0 {                                                                 // Wait for TASK_FINISHED update to be sent (async)
+	}
+	assert.Equal(s.T(), *mesos.TASK_FINISHED.Enum(), *pullFirstUpdate(s.executor.UnackedUpdates).Status.State) // Should be a TASK_FINISHED update
 
-	containerTask := s.executor.ContainerTasks[s.taskInfo.GetTaskID()]
-	assert.Equal(s.T(), "fakeContainerID", containerTask.ContainerID) // Should be equal to the container ID
+	monkey.UnpatchAll()
```

Comment:
WHy not using a defer to unpatchAll ?

Reply:
This call was not the last one of the test before. I forgot to move it, (done).